core/src/modules/ops/destination.ts

```diff
@@ -924,37 +922,36 @@ export namespace DestinationOps {
   }> {
     const _exports: Export[] = [];
     const locks: Awaited<ReturnType<typeof getLock>>[] = [];
+    const exportsWithoutChanges = givenExports.filter((e) => !e.hasChanges);
+    const exportsWithChanges = givenExports.filter((e) => e.hasChanges);
```

Comment:
Can you help me understand this change?  `.filter` is marginally slower than a `for` loop, and now we have multiple `.filter`s and then loops through the results of those instead of the single `for` below.  It looks cleaner, but I would think performance would be slower.

Reply:
You are right! I should be making both of theses new arrays in one loop, rather than 2.  I was mostly concerned with the order-of-magnitude change to reduce the number of SQL statements +network requests made, but this should be better.

Reply:
The SQL query redundancy should be further reduced by #2943 so even faster ⏩ ⏩ ⏩

Reply:
That explains the merge conflicts :D